Big_Integer/main.cpp

```diff
@@ -0,0 +1,476 @@
+        }
+
+        normalize(a, b);
+        string leftA, rightA;
+        string leftB, rightB;
+        _break(a, leftA, rightA);
+        _break(b, leftB, rightB);
+        string p1 = multiply(leftA, leftB);
+        string p2 = multiply(rightA, rightB);
+        string a_sum = add(leftA, rightA);
+        string b_sum = add(leftB, rightB);
+        string p3 = multiply(a_sum, b_sum);
+        string delta31 = subtract(p3, p1);
+        string p4 = subtract(delta31, p2);
+        insertTrailingZeros(p1, (int)a.size());
+        insertTrailingZeros(p4, (int)a.size() >> 1);
+        string sum= add(p2, p4);
+        return add(p1, sum);
+    }
+
+
+
+    string _modulo(string a, string b) { // modulo operation of two integers(strings)
+        string result = "";
+        string current = "";
+        for (char ch: a) {
+            current += ch;
+            if (_compare(current, b) != -1) {
+                int wheel = 0;
+                while (_compare(current, b) != -1) {
+                    wheel ++;
+                    current = subtract(current, b);
+                }
+                result += (char)(wheel + '0');
+            }
+        }
+        reverse(current.begin(), current.end());
+        while ((int)current.size() > 1 && current.back() == '0') {
+            current.pop_back();
+        }
+        reverse(current.begin(), current.end());
+        return current;
+    }
+    string fast_modulo(string a, int b) { // faster modulo function
+        string result = "";
+        int current = 0;
+        for (char ch: a) {
+            current = (current << 1) + (current << 3) + ch - '0';
+            if (current >= b) {
+                result += (char)(current / b + '0');
+                current %= b;
+            }
+        }
+        return to_string(current);
+    }
+
+    string modulo(string a, string b) { // Day 5: modulo of two integers
+        if ((int)b.size() > 9) {
+            return _modulo(a, b);
+        } else {
+            return fast_modulo(a, stoi(b));
+        }
+    }
+
+
+//OPERATORS
+//OVERLOADING
+//DAY 5 almost completed
+bool operator == (Integer passed) { //Day 4: equal to operator
+        return (sign == passed.get_sign() || isZero()) && value == passed.getValue();
+    }
+
+    bool operator < (Integer passed) { // less than operator
+        if (sign !=passed.get_sign()) {
+            return sign < passed.get_sign();
+        }
+        int length_passed = passed.length();
+        int length = this->length();
+
+        if (length < length_passed) {
+            return sign == 1;
+        } else if (length > length_passed) {
+            return sign == -1;
+        }
+        for (int i = 0; i < length; i ++) {
+            if (value[i] - '0' != passed.digitAt(i)) {
+                return value[i] - '0' < passed.digitAt(i) && sign == 1;
+            }
+        }
+        return false;
+    }
+
+    bool operator <= (Integer passed) { // less or equal operator
```

Comment:
Codacy found an issue: [Function parameter 'passed' should be passed by const reference.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6290669)